flybywiresim/a32nx
# fix: pushback without command

This is a fix for the case where the tug started an unintentional push back by calling the tug first from the ground tab in the EFB and then switching to another tab in the EFB while the tug was still on its way.

Fixes #5315

## Summary of Changes

To prevent the tug from starting a push back on its arrival a timer was used to continually set the PushBackWait simvar. This timer got cancelled by the useEffect that started it when the Ground component got unloaded by selecting some other tab in the EFB for instance. 

The handle for the timer is now kept in the redux store and used to clear the timer when the tug is dismissed again.
## Screenshots (if necessary)



## References





## Additional context



Discord username (if different from GitHub):

## Testing instructions

1. Select the ground tab in the EFB and click the tug button to request the tug.
2. While the tug is on its way select some other tab in the EFB so that the ground tab is unloaded.
3. Wait for the tug to arrive an connect. I should not start with a push back at this stage.
4. Select the ground tab again and click one of the push back arrows to start a push back. Push back is expected to start.
5. Click the push back stop button. Push back is expected to stop.
6. Click the request tug button again. The tug is expected to be dismissed.


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
